app/src/org/commcare/activities/CommCareSetupActivity.java

```diff
@@ -426,6 +486,12 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                 setResult(RESULT_CANCELED);
                 finish();
                 return;
+            case ConnectManager.CONNECTID_REQUEST_CODE :

```

Comment:
This should be handled via ConnectManager.handleFinishedActivity instead (called in the default case of this switch)